src/components/Timeline/index.tsx

```diff
@@ -25,22 +39,75 @@ const Timeline: React.FunctionComponent<TimelineProps> = ({
   updates,
 }) => {
   const [expanded, setExpanded] = React.useState<boolean>(!!showAll);
-
-  const sorted = useMemo(
-    () =>
-      updates.sort(
-        (u0, u1) =>
-          new Date(u1.timestamp).getTime() - new Date(u0.timestamp).getTime()
-      ),
-    [updates]
-  );
+  const {
+    filteredUpdates,
+    updateTypeFilter,
+    setUpdateTypeFilter,
+    groupedUpdates,
+  } = useFilterUpdates(updates);
 
   return (
     <Fade appear in timeout={1000}>
       <Grid container direction="column" spacing={5}>
+        <Grid item sm={6} xl={4} xs={12}>
+          {/* Filter timeline select */}
+          <Select
+            fullWidth
+            onChange={(event) =>
+              setUpdateTypeFilter(
+                event.target.value as UPDATE_TYPE_FILTER_OPTIONS
+              )
+            }
+            renderValue={(value) => (
+              <Typography color="secondary">
+                <FormattedMessage
+                  id="misc.timeline.filter.filterSelectLabel"
+                  values={{
+                    filter: (
+                      <FormattedMessage
+                        id={`misc.timeline.filter.byType.${value}`}
+                      />
+                    ),
+                  }}
+                />
+              </Typography>
+            )}
+            value={updateTypeFilter}
+            variant="outlined"
+          >
+            {Object.values(UPDATE_TYPE_FILTER_OPTIONS).map((type) => {
+              return (
+                <MenuItem
+                  key={type}
+                  disabled={groupedUpdates[type].length === 0}
+                  value={type}
+                >
+                  <FormattedMessage
+                    id={`misc.timeline.filter.byType.${type}`}
+                  />
+                </MenuItem>
+              );
+            })}
```

Comment:
Instead of using the enum directly, maybe `useFilterUpdates()` could return an array of options?

Reply:
Not a bad idea. Will do.